pkg/hive/cell/config.go

```diff
@@ -104,8 +104,24 @@ func decoderConfig(target any) *mapstructure.DecoderConfig {
 		Result:           target,
 		WeaklyTypedInput: true,
 		DecodeHook: mapstructure.ComposeDecodeHookFunc(
+			// To unify the splitting of fields of a []string field across the input coming
+			// from environment, configmap and pflag (command-line), we first split a string
+			// (env/configmap) by comma, and then for all input methods we split a single
+			// value []string by whitespace. Thus the following all result in the same slice:
+			//
+			// --string-slice=foo,bar,baz
+			// --string-slice="foo bar baz"
+			// CILIUM_STRING_SLICE="foo,bar,baz"
+			// CILIUM_STRING_SLICE="foo bar baz"
+			// /.../configmap/string_slice: "foo bar baz"
+			// /.../configmap/string_slice: "foo,bar,baz"
+			//
+			// If both commas and whitespaces are present the commas take precedence:
+			// "foo,bar baz" => []string{"foo", "bar baz"}
+			mapstructure.StringToSliceHookFunc(","), // string->[]string is split by comma
+			fixupStringSliceHookFunc,                // []string of length 1 is split again by whitespace
```

Comment:
Nit regarding the function name (`fixupStringSliceHookFunc`). The word 'fixup' is relatively meaningless, maybe something like `stringElemToSliceByWhitespace` could work?

Reply:
Thanks, I'll fix it if I need to rebase/rerun CI, otherwise will merge.